Pokemon_Data/matchup_scoring.py

```diff
@@ -222,7 +222,7 @@ def select_best_move(attacker: Pokemon, defender: Pokemon, teammates: Dict[str,
     """Return the index of the move that the attacker should use against the defender."""
     best_score = -100
     best_index = 0
-    for i in range(len(attacker.moves)):

```

Comment:
This approach should now work with Ditto having its 5th move removed after copying it from a boss.